service-print/src/main/java/org/oskari/print/loader/CommandLoadImageWMTS.java

```diff
@@ -190,10 +198,51 @@ private TileMatrix getTileMatrix(TileMatrixSet tms) throws IllegalArgumentExcept
     }
 
     private TileMatrixSet getTileMatrixSet() throws IllegalArgumentException {
+        List<TileMatrixSet> possibleTileMatrixSets = new ArrayList<>();
+
         for (TileMatrixSet tms : capabilities.getTileMatrixSets()) {
-            if (srs.equals(ProjectionHelper.shortSyntaxEpsg(tms.getCrs()))) {
-                return tms;
+            String key = tms.getTileMatrixMap().keySet().iterator().next();
+            possibleTileMatrixSets.add(tms);
+        }
+
+        if (possibleTileMatrixSets.isEmpty()) {
+            throw new NullPointerException("Could not find any TileMatrixSets");
+        }
+
+        if (possibleTileMatrixSets.size() == 1) {
+            return possibleTileMatrixSets.get(0);
+        }
+
+        HashMap<String, String> alternativeTileMatrixSets = new HashMap<>();
+        JSONObject useThisTileMatrixSetInstead = layerService.find(layer.getId()).getOptions().optJSONObject("useThisAlternativeTileMatrixSet");
+
+        if (useThisTileMatrixSetInstead == null) {
+            alternativeTileMatrixSets.put(srs, srs);
+        } else if (useThisTileMatrixSetInstead.length() > 0) {
+            Iterator iterator = useThisTileMatrixSetInstead.keys();
+            while (iterator.hasNext()) {
+                String nextKey = (String) iterator.next();
+                alternativeTileMatrixSets.put(nextKey, useThisTileMatrixSetInstead.optString(nextKey));
             }
+
+        }
+
+        TileMatrixSet setToReturn = null;
```

Comment:
I can't really understand what the rest of the function is trying to do. Let's think about what the current state here is:
- We have a `List<TileMatrixSet> possibleTileMatrixSets` all supporting the correct `srs`
- We have a `JSONObject useThisTileMatrixSetInstead` that maps the `srs`to the correct `TileMatrixSet#getIdentifier()` - To get the identifier we should use to use we call `useThisTileMatrixSetInstead.optString(srs);`
- The identifier we got from the `JSONObject` might be null, which we should handle accordingly, either by defaulting to `possibleTileMatrixSets.get(0)` or by throwing an exception
- Now we search the possible `possibleTileMatrixSets` for the one with the exact same `#getIdentifier()` value, for example by for-each looping the `possibleTileMatrixSets` and returning the list entry if we determine the identifiers match.
- If we can't find a `TileMatrixSet` from the `possibleTileMatrixSets` with the correct identifier we should throw an exception since this is clearly a configuration error, for example the layer tells us to use `ETRS-GK25` when srs is `EPSG:3879` but the WMTS service doesn't have a TileMatrixSet which both has the identifier `ETRS-GK25` and supports `EPSG:3879`.